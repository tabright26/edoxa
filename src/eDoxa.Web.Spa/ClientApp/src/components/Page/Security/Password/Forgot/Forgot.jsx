import React from "react";
import { Card, CardBody, Col, Container, Row } from "reactstrap";
import SecurityPassword from "../../../../../forms/Security/Password";
import withForgotPassword from "../../../../../containers/App/Security/withForgotPassword";

const ForgotPassword = ({ actions }) => (
  <div className="app flex-row align-items-center">
    <Container>
      <Row className="justify-content-center">
        <Col md="9" lg="7" xl="6">
          <Card className="mx-4">
            <CardBody className="p-4">
              <h1>Forgot Password</h1>
              <p className="text-muted">Forgot your account password</p>
              <SecurityPassword.Forgot onSubmit={fields => actions.forgotPassword(fields)} />
            </CardBody>
          </Card>
        </Col>
      </Row>
    </Container>
  </div>
);

export default withForgotPassword(ForgotPassword);
