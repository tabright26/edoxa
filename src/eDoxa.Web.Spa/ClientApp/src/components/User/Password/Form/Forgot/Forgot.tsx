import React, { FunctionComponent } from "react";
import { FormGroup, Form } from "reactstrap";
import { Field, reduxForm, InjectedFormProps, FormErrors } from "redux-form";
import Button from "components/Shared/Button";
import Input from "components/Shared/Input";
import { FORGOT_USER_PASSWORD_FORM } from "utils/form/constants";
import { compose } from "recompose";
import FormValidation from "components/Shared/Form/Validation";
import { throwSubmissionError } from "utils/form/types";
import { forgotUserPassword } from "store/actions/identity";
import { EMAIL_REQUIRED, EMAIL_INVALID, EMAIL_REGEXP } from "validation";
import { AxiosActionCreatorMeta } from "utils/axios/types";
import { push } from "connected-react-router";
import { toastr } from "react-redux-toastr";

interface FormData {
  email: string;
}

interface OutterProps {}

type InnerProps = InjectedFormProps<FormData, Props>;

type Props = InnerProps & OutterProps;

const CustomForm: FunctionComponent<Props> = ({ handleSubmit, error }) => (
  <Form onSubmit={handleSubmit}>
    {error && <FormValidation error={error} />}
    <Field
      type="text"
      name="email"
      label="Email"
      formGroup={FormGroup}
      component={Input.Text}
    />
    <FormGroup className="mb-0 d-flex">
      <Button.Submit className="w-25">Send</Button.Submit>
      <Button.Link className="ml-auto my-auto" to="/authentication/login">
        Return to login page
      </Button.Link>
    </FormGroup>
  </Form>
);

const enhance = compose<InnerProps, OutterProps>(
  reduxForm<FormData, Props>({
    form: FORGOT_USER_PASSWORD_FORM,
    onSubmit: async (values, dispatch) => {
      try {
        return await new Promise((resolve, reject) => {
          const meta: AxiosActionCreatorMeta = { resolve, reject };
          dispatch(forgotUserPassword(values, meta));
        });
      } catch (error) {
        throwSubmissionError(error);
      }
    },
    onSubmitSuccess: (_result, dispatch) => {
      dispatch(push("/"));
      setTimeout(function() {
        toastr.success(
          "Email sent",
          "We have sent you a link to reset your password by email.",
          { timeOut: 7500 }
        );
      }, 2500);
    },
    validate: values => {
      const errors: FormErrors<FormData> = {};
      if (!values.email) {
        errors.email = EMAIL_REQUIRED;
      } else if (!EMAIL_REGEXP.test(values.email)) {
        errors.email = EMAIL_INVALID;
      }
      return errors;
    }
  })
);

export default enhance(CustomForm);
