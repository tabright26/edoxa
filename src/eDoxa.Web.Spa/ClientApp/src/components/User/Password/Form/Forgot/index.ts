import ForgotUserPasswordForm from "./Forgot";

export default ForgotUserPasswordForm;
