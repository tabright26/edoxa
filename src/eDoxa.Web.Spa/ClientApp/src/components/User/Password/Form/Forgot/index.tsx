import React, { FunctionComponent } from "react";
import { FormGroup, Form, Button as BsButton } from "reactstrap";
import { Field, reduxForm, InjectedFormProps, FormErrors } from "redux-form";
import Button from "components/Shared/Button";
import Input from "components/Shared/Input";
import { FORGOT_USER_PASSWORD_FORM } from "utils/form/constants";
import { compose } from "recompose";
import { ValidationSummary } from "components/Shared/ValidationSummary";
import { throwSubmissionError } from "utils/form/types";
import { forgotUserPassword } from "store/actions/identity";
import {
  EMAIL_REQUIRED,
  EMAIL_INVALID,
  EMAIL_REGEXP
} from "utils/form/validators";
import { AxiosActionCreatorMeta } from "utils/axios/types";
import { push } from "connected-react-router";
import { toastr } from "react-redux-toastr";
import { LinkContainer } from "react-router-bootstrap";

interface FormData {
  email: string;
}

interface OutterProps {}

type InnerProps = InjectedFormProps<FormData, Props>;

type Props = InnerProps & OutterProps;

const CustomForm: FunctionComponent<Props> = ({
  handleSubmit,
  error,
  submitting
}) => (
  <Form onSubmit={handleSubmit}>
    <ValidationSummary error={error} />
    <Field
      type="text"
      name="email"
      placeholder="Email"
      formGroup={FormGroup}
      component={Input.Text}
    />
    <FormGroup className="mb-0 d-flex">
      <Button.Submit loading={submitting} className="w-25">
        Send
      </Button.Submit>
      <LinkContainer to="/authentication/login">
        <BsButton className="ml-auto my-auto" color="link" size="sm">
          Return to login page
        </BsButton>
      </LinkContainer>
    </FormGroup>
  </Form>
);

const enhance = compose<InnerProps, OutterProps>(
  reduxForm<FormData, Props>({
    form: FORGOT_USER_PASSWORD_FORM,
    onSubmit: async (values, dispatch) => {
      try {
        return await new Promise((resolve, reject) => {
          const meta: AxiosActionCreatorMeta = { resolve, reject };
          dispatch(forgotUserPassword(values, meta));
        });
      } catch (error) {
        throwSubmissionError(error);
      }
    },
    onSubmitSuccess: (_result, dispatch) => {
      dispatch(push("/"));
      setTimeout(function() {
        toastr.success(
          "Email sent",
          "We have sent you a link to reset your password by email."
        );
      }, 1500);
    },
    validate: values => {
      const errors: FormErrors<FormData> = {};
      if (!values.email) {
        errors.email = EMAIL_REQUIRED;
      } else if (!EMAIL_REGEXP.test(values.email)) {
        errors.email = EMAIL_INVALID;
      }
      return errors;
    }
  })
);

export default enhance(CustomForm);
