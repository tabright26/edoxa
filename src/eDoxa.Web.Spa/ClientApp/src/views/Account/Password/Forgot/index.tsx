import React, { FunctionComponent } from "react";
import { Card, CardBody } from "reactstrap";
import PasswordForm from "components/Service/Identity/Account/Password/Form";

const Forgot: FunctionComponent = () => (
  <Card className="mx-4">
    <CardBody className="p-4">
      <h1>Forgot password</h1>
      <p className="text-muted">
        Enter your email address. You will receive a link to reset your
        password.
      </p>
      <PasswordForm.Forgot />
    </CardBody>
  </Card>
);

export default Forgot;
