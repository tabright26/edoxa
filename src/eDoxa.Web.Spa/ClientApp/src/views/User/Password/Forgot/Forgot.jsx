import React from "react";
import { Card, CardBody } from "reactstrap";
import SecurityPassword from "forms/User/Password";
import withUserContainer from "containers/withUserContainer";

const ForgotPassword = ({ actions }) => (
  <Card className="mx-4">
    <CardBody className="p-4">
      <h1>Forgot Password</h1>
      <p className="text-muted">Forgot your account password</p>
      <SecurityPassword.Forgot onSubmit={fields => actions.forgotPassword(fields)} />
    </CardBody>
  </Card>
);

export default withUserContainer(ForgotPassword);
