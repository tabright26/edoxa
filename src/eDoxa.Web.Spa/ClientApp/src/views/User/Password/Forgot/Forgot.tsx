import React, { FunctionComponent } from "react";
import { Card, CardBody } from "reactstrap";
import UserPasswordForm from "components/User/Password/Form";

const ForgotUserPassword: FunctionComponent = () => (
  <Card className="mx-4">
    <CardBody className="p-4">
      <h1>Forgot password</h1>
      <p className="text-muted">
        Enter your email address. You will receive a link to reset your
        password.
      </p>
      <UserPasswordForm.Forgot />
    </CardBody>
  </Card>
);

export default ForgotUserPassword;
