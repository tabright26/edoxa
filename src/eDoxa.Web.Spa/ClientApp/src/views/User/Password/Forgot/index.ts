import ForgotUserPassword from "./Forgot";

export default ForgotUserPassword;
