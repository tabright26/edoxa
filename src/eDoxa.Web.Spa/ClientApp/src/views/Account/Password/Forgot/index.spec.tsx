import React from "react";
import renderer from "react-test-renderer";
import Forgot from "../Forgot";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import store from "store";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <Forgot />
        </MemoryRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
