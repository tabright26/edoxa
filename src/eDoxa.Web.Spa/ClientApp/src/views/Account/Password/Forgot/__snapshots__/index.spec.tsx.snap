// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mx-4 card"
>
  <div
    className="p-4 card-body"
  >
    <h1>
      Forgot password
    </h1>
    <p
      className="text-muted"
    >
      Enter your email address. You will receive a link to reset your password.
    </p>
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <input
          autoComplete="email"
          className="form-control-sm form-control"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="Email"
          type="text"
          value=""
        />
        <div
          className="invalid-feedback"
        />
      </div>
      <div
        className="mb-0 d-flex form-group"
      >
        <button
          aria-label={null}
          className="w-25 btn btn-primary"
          disabled={null}
          onClick={[Function]}
          type="submit"
        >
          Send
        </button>
        <button
          aria-label={null}
          className="p-0 ml-auto my-auto btn btn-link btn-lg"
          disabled={null}
          onClick={[Function]}
          type="button"
        >
          <small
            className="null"
          >
             
            Return to login page
          </small>
        </button>
      </div>
    </form>
  </div>
</div>
`;
